### PR TITLE
feat: Adding new dataflow pipelines to import HBase snapshot and import validation

### PR DESCRIPTION
**Pull request by vermas2012**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable-hbase/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

File: .kokoro/nightly/integration-beam.cfg [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Format: //devtools/kokoro/config/proto/build.proto
```
Comment (igorbernstein2) [inline .kokoro/nightly/integration-beam.cfg:1]: @kolea2 what else needs to be setup to run these ITs from this branch?

File: bigtable-dataflow-parent/bigtable-beam-import/README.md [hunk 2]
```diff
@@ -4,8 +4,8 @@ This project supports importing and exporting HBase Sequence Files to Google Clo
 Cloud Dataflow.
 
 ## Instructions
-
-Download [the import/export jar](http://search.maven.org/remotecontent?filepath=com/google/cloud/bigtable/bigtable-beam-import/1.1.0/bigtable-beam-import-1.1.0-shaded.jar), which is an aggregation of all required jars.
+[//]: # ({x-version-update-start:bigtable-dataflow-parent:released})
+Download [the import/export jar](http://search.maven.org/remotecontent?filepath=com/google/cloud/bigtable/bigtable-beam-import/1.14.1/bigtable-beam-import-1.14.1-shaded.jar), which is an aggregation of all required jars.
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-beam-import/README.md:8]: here and everywhere else, the version looks stale

File: bigtable-dataflow-parent/bigtable-beam-import/pom.xml [hunk 3]
```diff
@@ -178,6 +210,29 @@ limitations under the License.
...
+      <version>1.0.1</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.google.cloud</groupId>
+      <artifactId>google-cloud-bigtable-emulator</artifactId>
+      <version>0.124.0</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-beam-import/pom.xml:231]: Please move this out of the test section

File: bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java [hunk 4]
```diff
@@ -72,6 +73,19 @@ public static CloudBigtableTableConfiguration BuildImportConfig(ImportOptions op
     return builder.build();
   }
 
+  /** Builds CloudBigtableTableConfiguration from input runtime parameters for import job. */
+  public static CloudBigtableTableConfiguration BuildSyncTableConfig(SyncTableOptions opts) {
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java:77]: in a followup PR please fix naming convention

File: pom.xml [hunk 5]
```diff
@@ -263,4 +263,26 @@ limitations under the License.
...
+          <releases>
+            <enabled>false</enabled>
+          </releases>
+          <snapshots>
+            <enabled>true</enabled>
+          </snapshots>
+        </repository>
+      </repositories>
+    </profile>
+  </profiles>
```
Comment (igorbernstein2) [inline pom.xml:287]: why is this necessary?

File: pom.xml [hunk 6]
```diff
@@ -67,24 +67,24 @@ limitations under the License.
...
-    <beam-guava.version>20.0</beam-guava.version>
-    <beam-auto-value.version>1.7</beam-auto-value.version>
+    <guava.version>30.1-android</guava.version>
+    <gcs-guava.version>29.0-jre</gcs-guava.version>
+    <beam-auto-value.version>1.7.4</beam-auto-value.version>
     <beam-grpc.version>1.29.0</beam-grpc.version>
 
     <!-- Benchmarks related dependencies -->
-    <jmh.version>1.23</jmh.version>
+    <jmh.version>1.27</jmh.version>
```
Comment (igorbernstein2) [inline pom.xml:87]: why all of the version upgrades?

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml [hunk 7]
```diff
@@ -14,13 +14,16 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
 
   <parent>
     <groupId>com.google.cloud.bigtable</groupId>
     <artifactId>bigtable-hbase-1.x-parent</artifactId>
-    <version>1.19.3-SNAPSHOT</version> <!-- {x-version-update:bigtable-hbase-1x-parent:current} -->
+    <version>1.19.3-SNAPSHOT
+    </version> <!-- {x-version-update:bigtable-hbase-1x-parent:current} -->
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml:26]: please keep this on the same line

**Review by igorbernstein2 — CHANGES_REQUESTED** [review]
I'm assuming that all of the java code is the same as the code that was already reviewed in master, so I'm mainly focused on the poms

**Comment by vermas2012** [comment]
> I'm assuming that all of the java code is the same as the code that was already reviewed in master, so I'm mainly focused on the poms

Yes.

File: bigtable-dataflow-parent/bigtable-beam-import/README.md (hunk 2 above)
Comment (vermas2012) [reply bigtable-dataflow-parent/bigtable-beam-import/README.md:8]: Billy has a PR in review that revamps all the readme. We will have the PR merged in both places. I will update the versions here.

File: .kokoro/nightly/integration-beam.cfg (hunk 1 above)
Comment (kolea2) [reply .kokoro/nightly/integration-beam.cfg:1]: This is sufficient - it looks like the presubmit ran (though failed due to test errors)

File: bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java (hunk 4 above)
Comment (vermas2012) [reply bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java:77]: ack

File: pom.xml (hunk 6 above)
Comment (vermas2012) [reply pom.xml:87]: These upgrades were done as part of HBaseSnapshotImport work, I am bringing them as is. I don't know if the import job breaks if we downgrade them.

File: pom.xml (hunk 5 above)
Comment (vermas2012) [reply pom.xml:287]: I am not sure. I think Cheng added it for enabling integ tests. @kolea2 do you know?

Comment (kolea2) [reply pom.xml:287]: You can revert this. This is getting pulled in from https://github.com/googleapis/java-bigtable-hbase/pull/2535, which isn't configured to run on this branch.

**Review by kolea2 — CHANGES_REQUESTED** [review]

**Comment by kolea2** [comment]
Also, please rename this PR using conventional commits with what you would want it to appear as in the release notes.

File: pom.xml (hunk 5 above)
Comment (kolea2) [reply pom.xml:287]: @vermas2012 friendly ping :)

File: pom.xml [hunk 8]
```diff
@@ -67,24 +67,24 @@ limitations under the License.
     <hbase.version.1>1.4.12</hbase.version.1>
     <hbase.version.2>2.2.3</hbase.version.2>
     <hbase.version>${hbase.version.1}</hbase.version>
-    <hbase1-hadoop.version>2.7.4</hbase1-hadoop.version>
+    <hbase1-hadoop.version>2.10.1</hbase1-hadoop.version>
```
Comment (igorbernstein2) [inline pom.xml:70]: I dont think this PR should upgrade the hadoop version

File: pom.xml [hunk 9]
```diff
@@ -67,24 +67,24 @@ limitations under the License.
...
+    <mockito.version>3.7.7</mockito.version>
     <!-- TODO: check if commons-codec was transitively updated to 1.13 and okhttp was updated to 2.7.5 when upgrading-->
     <beam.version>2.24.0</beam.version>
     <!-- referred from bigtable-beam-import and bigtable-emulator -->
-    <guava.version>30.0-android</guava.version>
-    <beam-guava.version>20.0</beam-guava.version>
-    <beam-auto-value.version>1.7</beam-auto-value.version>
+    <guava.version>30.1-android</guava.version>
+    <gcs-guava.version>29.0-jre</gcs-guava.version>
+    <beam-auto-value.version>1.7.4</beam-auto-value.version>
```
Comment (igorbernstein2) [inline pom.xml:83]: I'm a bit concerned about these version upgrades. Why are they necessary?

File: pom.xml (hunk 5 above)
Comment (vermas2012) [reply pom.xml]: removed.

**Review by igorbernstein2 — APPROVED** [review]
lgtm
